src/app/shared/services/project/project.service.ts

```diff
@@ -117,4 +117,5 @@ export interface PageableProjectsResponse {
 export interface ProjectWallet {
     project: Project;
     wallet: Wallet;
+    organization: Organization;
```

Comment:
what is that? how I see inside the [documentation](https://staging.ampnet.io/docs/project/#_get_all_projects_for_organization), there is no `organization` property in `project_wallet` response. how does this work? do I miss something?

Reply:
We need to have this Organization property in ProjectWallet so we can retrieve the organization UUID and the 
organization name, which we display in single-offer-item next to Published by. 
Please check this [documentation](https://staging.ampnet.io/docs/project/#_get_active_projects) and also what was written in this [task](https://dev.azure.com/ampnetio/AMPnet%20Crowdfunding/_workitems/edit/670/).

Reply:
Yes, it's obvious why we need the organization info for the particular project, but I've linked you the endpoint in the documentation where we use the same interface (ProjectWallet), but there is no organization in the response. So, even though we still don't use it in the "project for organization" response, it is nullable or guys on the backend forgot to put the organization there. Think about interfaces as a contract (in some languages, interfaces are called contracts). It is our type-safe contract between our beliefs what our backend is returning. Somebody in the future will be highly convinced that there is always an organization here, will not set a null safety operator and the thing will break in the runtime.

Let it stay like this, but I don't like it, so let me clarify why.

We previously had routes for active projects and projects for organizations that had a collection of products called "products", so we mapped every single item as `Product` on our side. 
There was a need on that endpoint to return **blockchain expensive** wallet information about the project wallet. It was inconvenient to return this expensive and full wallet information inside a project response, so we added a `wallet` response alongside with `project` inside the collection. That way, we renamed the collection as "projects_wallets" to indicate the collection, with a single item called "ProjectWallet" on our side.

Now, we needed simple (not expressive nor full) info about the project organization, so the organization was added inside the "projects_wallets" response, thus breaking the meaning of ProjectWallet because it should be called ProjectWalletOrganization now, but it seems we are going in a wrong direction.

What I suggest is that a project, wherever we fetch it, should have at least basic information about its relatives (wallet, organization, etc.). When I say basic information, I mean not full information about the organization (e.g. with all their documents) nor full blockchain response of the project wallet, but just basic information (org: UUID, name; wallet: activation_data, hash?) or whatever we say is basic. This way, whenever we fetch the project from the API, we have a connection about project relations; we have the name and UUID of the project organization so we can set a link to the organization page. we have activation data (or hash) to get a full response from the wallet from the blockchain. 

Just mentioning @githinho to be mentioned 😄